architect.yml

```diff
@@ -1,68 +1,81 @@
-    required: false
-  api_db_name:
-    description: Name of the database used by the API
-    default: test_database
-  root_db_user:
-    description: Root user to assign to the generated database
-    default: root
-  root_db_pass:
-    description: Root password to assign to the generated database
+  db_user:
+    description: Root user to assign to the component's database
     default: architect
+  db_pass:
+    description: Root password to assign to the component's database
+    default: secret
+  db_name:
+    description: Name of the database the component will store content in
+    default: api-db
+  db_port:
+    description: Port for the database
+    default: 5432
   api_port:
     description: Port for api
     default: 3000
 
 services:
-  api-db:
-    image: mariadb:latest
-    interfaces:
-      database:
-        host: ${{ secrets.api_db_host }}
-        port: 3306
-        username: ${{ secrets.root_db_user }}
-        password: ${{ secrets.root_db_pass }}
-        protocol: mysql
-        path: /${{ secrets.api_db_name }}
-    environment:
-      MYSQL_ROOT_PASSWORD: ${{ secrets.root_db_pass }}
-      MYSQL_DATABASE: ${{ secrets.api_db_name }}
-
+  ### This service is the actual Nest.js component ###
   api:
-    depends_on:
-      - api-db
+    # Build context is the location of your application relative to the architect.yml file
     build:
-      context: ./backend
+      context: .
       args:
         NODE_ENV: production
+    depends_on:
+      - api-db
     interfaces:
       main:
         port: ${{ secrets.api_port }}
         ingress:
           subdomain: api
+    # The liveness_probe makes sure the application is healthy - if not, it will re-start the application
     liveness_probe:
-      command: curl --fail localhost:3000/items
+      command: curl --fail localhost:3000
       interval: 30s
       failure_threshold: 3
+    # Set environment variables for your application here
     environment:
       PORT: ${{ secrets.api_port }}
-      DB_ADDR: ${{ services.api-db.interfaces.database.url }}
+      PG_USER: ${{ secrets.db_user }}
+      PG_PASSWORD: ${{ secrets.db_pass }}
+      PG_DATABASE: ${{ secrets.db_name }}
+      DB_HOST: ${{ services.api-db.interfaces.main.host }}
+      DB_PORT: ${{ services.api-db.interfaces.main.port }}
+    # Debug block is used to enable hot-reloading to make real-time changes to an
+    #   application while running `architect dev`. For more info, see -
+    #   https://docs.architect.io/components/local-configuration/#hot-reloading
     debug:
       command: npm run start:debug
       build:
         args:
           NODE_ENV: development
       volumes:
         src:
-          mount_path: /usr/src/app/src
-          host_path: ./backend/src
+          # The `WORKDIR` defined in the Dockerfile (/usr/src/app)
+          mount_path: /usr/src/app
+          # Path to source code relative to the architect.yml file
+          host_path: .
```

Comment:
The reason why I couldn't run this was because this specifies that the entire host directory should be loaded to the `WORKDIR`. That means that whatever was in the `WORKDIR` when the Docker image was built will be completely replaced by whatever is in the `.` directory on the host machine. I didn't have the node modules installed, so the app couldn't run. Having node modules installed to the host machine shouldn't be required to run a container though. What should be mounted here are only the directories that will be getting updated on the host, and where the changes will be needed in the container. ex

```yml
    volumes:
        src:
          # The `WORKDIR` defined in the Dockerfile (/usr/src/app) + src
          mount_path: /usr/src/app/src
          # Path to source code relative to the architect.yml file
          host_path: ./src
        public:
          # The `WORKDIR` defined in the Dockerfile (/usr/src/app) + public
          mount_path: /usr/src/app/public
          # Path to public code relative to the architect.yml file
          host_path: ./public
```